airflow/hooks/python_virtualenv.py

```diff
@@ -0,0 +1,350 @@
+        system_site_packages: None | bool = None,
+        pip_install_options: list[str] | None = None,
+        index_urls: None | Collection[str] | str = None,
+        venv_cache_path: None | os.PathLike[str] = None,
+        serializer: None | Literal["pickle", "cloudpickle", "dill"] = None,
+    ) -> None:
+        super().__init__()
+
+        self._venv_conn_id = venv_conn_id
+        self._requirements = requirements
+        self._python_version = python_version
+        self._system_site_packages = system_site_packages
+        self._pip_install_options = pip_install_options
+        self._index_urls = index_urls
+        self._venv_cache_path = venv_cache_path
+        self._serializer = serializer
+        self._conn = self.get_connection(venv_conn_id) if venv_conn_id else None
+
+    def get_conn(self) -> None:
+        pass
+
+    @cached_property
+    def extra(self) -> dict[str, Any]:
+        if self._conn:
+            return self._conn.extra_dejson
+        return {}
+
+    @property
+    def requirements(self) -> Iterable[str]:
+        if self._requirements:
+            return self._requirements
+        if self.extra.get("requirements"):
+            return [requirement.strip() for requirement in self.extra["requirements"].split("\n")]
+        return []
+
+    @property
+    def python_version(self) -> str | None:
+        return self._python_version or self.extra.get("python_version")
+
+    @property
+    def system_site_packages(self) -> bool:
+        if self._system_site_packages is not None:
+            return self._system_site_packages
+        return self.extra.get("system_site_packages") or True
+
+    @property
+    def pip_install_options(self) -> list[str]:
+        if self._pip_install_options:
+            return self._pip_install_options
+        if self.extra.get("pip_install_options"):
+            return self.extra["pip_install_options"].split()
+        return []
+
+    @property
+    def index_urls(self) -> Collection[str] | None:
+        if self._index_urls:
+            return self._index_urls
+        if self.extra.get("index_urls"):
+            return [index_url.strip() for index_url in self.extra["index_urls"].split(",")]
+        return []
+
+    @property
+    def venv_cache_path(self) -> Path | None:
+        if self._venv_cache_path:
+            return Path(self._venv_cache_path)
+        if self.extra.get("venv_cache_path"):
+            return Path(self.extra["venv_cache_path"])
+        return None
+
+    @property
+    def serializer(self) -> Literal["pickle", "cloudpickle", "dill"]:
+        if self._serializer:
+            return self._serializer
+        return self.extra.get("serializer") or "pickle"
+
+    def _requirements_list(self, exclude_cloudpickle: bool = False) -> list[str]:
+        """Prepare a list of requirements that need to be installed for the virtual environment."""
+        requirements = [str(dependency) for dependency in self.requirements]
```

Comment:
```suggestion
        requirements = [dependency for dependency in self.requirements]
```
requirements가 Iterable[str] 타입으로 반환하니까 프로퍼티쪽에서 처리해주는게 좋을듯요

Reply:
이거 기존 코드 그대로 복붙한건데, 뭔가 이유가 잇겟지요
그대로 가져갑니다
https://github.com/apache/airflow/blob/e30e1ee5401ab60e76eb34dd198385660a34faee/airflow/operators/python.py#L731-L744